modules/javafx.controls/src/main/java/javafx/scene/control/Labeled.java

```diff
@@ -818,6 +818,48 @@ public final BooleanProperty mnemonicParsingProperty() {
         return mnemonicParsing;
     }
 
+    /**
+     * Truncated read-only property indicates whether the text has been truncated
+     * when it cannot fit into the available width.
+     * <p>
+     * When truncated, the {@link #ellipsisStringProperty() ellipsis string}
+     * gets inserted in the place dictated by the
+     * {@link #textOverrun} property.
+     *
+     * @since 23
+     * @defaultValue false
+     */
+    private ObservableBooleanValue truncated;
+
+    public final ObservableBooleanValue truncatedProperty() {
```

Comment:
`ObservableBooleanValue` is not a property, so it's odd to use it as the return type of one. Usually `ReadOnlyBooleanProperty` is used. This implementation through binding is also unique for read-only properties. A look in `Node` and `Labeled` show a different way of implementing read-only properties.